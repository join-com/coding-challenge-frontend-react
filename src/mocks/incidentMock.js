const incidentMock = {
  id: 24134,
  title: "Stolen di Florino (red and white)",
  description: "",
  address: "Berlin, 10245, DE",
  occurred_at: 1402984800,
  updated_at: 1545551568,
  url: "https://bikewise.org/api/v1/incidents/24134",
  source: {
    name: "BikeIndex.org",
    html_url: "https://bikeindex.org/bikes/27159",
    api_url: "https://bikeindex.org/api/v1/bikes/27159"
  },
  media: {
    image_url: "https://bikeindex.s3.amazonaws.com/uploads/Pu/9564/large_dsc_1293.jpg",
    image_url_thumb: "https://bikeindex.s3.amazonaws.com/uploads/Pu/9564/small_dsc_1293.jpg"
  },
  location_type: null,
  location_description: null,
  type: "Theft",
  type_properties: null
};

export default incidentMock;
