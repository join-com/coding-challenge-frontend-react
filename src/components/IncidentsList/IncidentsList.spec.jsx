import React from "react";
import { shallow } from "enzyme";
import Component from "./IncidentsList";
import incidentMock from "../../mocks/incidentMock";

describe("IncidentsList component test", () => {
  const defaultProps = {
    items: [incidentMock],
    itemsPerPage: 10,
    changeShowItemsPerPage: jest.fn(),
    totalIncidentsLength: 23
  };

  const getWrapper = (testProps = {}) => {
    const props = { ...defaultProps, ...testProps };
    return shallow(<Component {...props} />);
  };

  it("should match snapshot ", () => {
    const wrapper = getWrapper();
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with empty items", () => {
    const wrapper = getWrapper({ items: [] });
    expect(wrapper).toMatchSnapshot();
  });
});
