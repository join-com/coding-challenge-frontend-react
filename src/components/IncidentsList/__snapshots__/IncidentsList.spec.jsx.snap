// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncidentsList component test should match snapshot  1`] = `
<div>
  <Card
    item={
      Object {
        "address": "Berlin, 10245, DE",
        "description": "",
        "id": 24134,
        "location_description": null,
        "location_type": null,
        "media": Object {
          "image_url": "https://bikeindex.s3.amazonaws.com/uploads/Pu/9564/large_dsc_1293.jpg",
          "image_url_thumb": "https://bikeindex.s3.amazonaws.com/uploads/Pu/9564/small_dsc_1293.jpg",
        },
        "occurred_at": 1402984800,
        "source": Object {
          "api_url": "https://bikeindex.org/api/v1/bikes/27159",
          "html_url": "https://bikeindex.org/bikes/27159",
          "name": "BikeIndex.org",
        },
        "title": "Stolen di Florino (red and white)",
        "type": "Theft",
        "type_properties": null,
        "updated_at": 1545551568,
        "url": "https://bikewise.org/api/v1/incidents/24134",
      }
    }
    key="24134"
  />
</div>
`;

exports[`IncidentsList component test should match snapshot with empty items 1`] = `<div />`;
